### PR TITLE
Bash Script for Mac OS X / Unix / Linux users

### PR DESCRIPTION
**Pull request by pythonmagic**
Created a bash Script for Mac OS X / Unix / Linux users

### CODE REVIEW

File: updateHosts.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+# This script will create in first running backup of ORIGINAL/CURRENT hosts file in hosts.skel file.
+# If hosts.skel file exists, then NEW copy with customized unified hosts file will be copied to proper path.
+# Next DNS Cache will be refreshed.
+# YOU NEED RUNNING THIS SCRIPT FILE IN COMMAND LINE PROMPT WITH ADMINISTRATOR PRIVILIGES
+#
+# Might have to change change execute for the script file
+# chmod 655 updaeHosts.sh
+#
+# sudo ./updaeHosts.sh
```
Comment (tomhoover) [inline updateHosts.sh]: sudo ./upda**t**eHosts.sh

File: updateHosts.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+# This script will create in first running backup of ORIGINAL/CURRENT hosts file in hosts.skel file.
+# If hosts.skel file exists, then NEW copy with customized unified hosts file will be copied to proper path.
+# Next DNS Cache will be refreshed.
+# YOU NEED RUNNING THIS SCRIPT FILE IN COMMAND LINE PROMPT WITH ADMINISTRATOR PRIVILIGES
+#
+# Might have to change change execute for the script file
+# chmod 655 updaeHosts.sh
```
Comment (tomhoover) [inline updateHosts.sh]: ./upda**t**eHosts.sh

**Comment by tomhoover** [comment]
Please ignore the last comment. It should have said that ./upda**t**eHosts.sh is misspelled in lines 8 & 10.

**Comment by StevenBlack** [comment]
@pythonmagic thanks for the pull request!  Very nice initiative.

Thanks for reviewing this Tom @tomhoover.

@pythonmagic can you make those changes?

**Comment by FadeMind** [comment]
1# `chmod 755` NOT `655`.
2# `./` will not working at all without EXEC bit (755) @tomhoover 
3# Better is using external HEADER
```
#!/usr/bin/env bash
```
4# Why not using `sudo` here directly in code??
5# After copy new hosts file good command will be clear DNS cache automatically.
6# Missing: error handling.
@StevenBlack

**Comment by pythonmagic** [comment]
Thank a lot for suggestion and correcting my typos.

i deliberately did not add DNS cache clear to the script since their are lot of variation in how to do it in unix and mac. 

macOS Sierra 10.12.0
sudo killall -HUP mDNSResponder

OSX 10.11.0
sudo killall -HUP mDNSResponder

OSX 10.10.4
sudo killall -HUP mDNSResponder

OSX 10.10.0 – 10.10.3
sudo discoveryutil mdnsflushcache

OSX 10.9  – 10.8 – 10.7
sudo killall -HUP mDNSResponder

OSX 10.5 – 10.6
sudo dscacheutil -flushcache

Windows
ipconfig /flushdns

Linux (depending on what you’re running)
/etc/init.d/named restart
/etc/init.d/nscd restart

**Comment by pythonmagic** [comment]
New Changed Script :)

#!/usr/bin/env bash
# This script will create in first running backup of ORIGINAL/CURRENT hosts file in hosts.skel file.
# If hosts.skel file exists, then NEW copy with customized unified hosts file will be copied to proper path.
# Next DNS Cache will be refreshed.
# YOU NEED RUNNING THIS SCRIPT FILE IN COMMAND LINE PROMPT WITH ADMINISTRATOR PRIVILIGES
#
# Might have to change change execute for the script file
# chmod 755 updateHosts.sh
#
# ./updaeHosts.sh OR sh updateHosts.sh

function flushDNS {
	# Find Mac version and cleares the DNS Cache
	MAJOR_MAC_VERSION=$(sw_vers -productVersion | awk -F '.' '{print $2}')
	if [ "$MAJOR_MAC_VERSION" -ge "10" ]
	then
		sudo killall -HUP mDNSResponder
	else
		sudo dscacheutil -flushcache
	fi
	echo "* * * DNS Cache Cleared * * *"
}

function updateHosts {
	echo "* * * Updating hosts * * *"
	python updateHostsFile.py -a >> /dev/null
	echo "* * * Copying hosts to /etc/hosts * * *"
	sudo cp hosts /etc/hosts
	flushDNS
}

if [ -f "/etc/hosts.skel" ]
then
	updateHosts
else
	echo "* * * Creating a Backup of current hosts file (/etc/hosts.skel) * * *"
	sudo cp /etc/hosts /etc/hosts.skel
	updateHosts
fi

**Comment by gfyoung** [comment]
@StevenBlack : At the very least, I think this can be closed?  Personally, not so sure if I would prefer to use a bash script over Python (extra maintenance), but at least we can close duplicates.

**Pull request closed without merging**